### PR TITLE
Flattened annotations

### LINKED ISSUE #11: RDF to JSON should use @annotation, where possible
The existing algorithm only uses embedded nodes.

### PR DESCRIPTION
**Pull request by gkellogg**
This adds a `createAnnotations` flag to the flattening algorithm, allowing annotation objects to be used instead of embedded nodes, where appropriate. It also uses this algorithm in Serialize RDF as JSON-LD.

Updates examples.

Fixes #11

***
<a href="https://pr-preview.s3.amazonaws.com/json-ld/json-ld-star/pull/13.html" title="Last updated on Feb 11, 2021, 10:16 PM UTC (051eaf0)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/json-ld/json-ld-star/13/04d8e1c...051eaf0.html" title="Last updated on Feb 11, 2021, 10:16 PM UTC (051eaf0)">Diff</a>

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -1341,43 +1363,89 @@ <h4>Rename Blank Nodes</h4>
...
-                  <li>Initialize an empty <a>array</a> <var>result</var>.</li>
-                  <li>For each <var>item</var> in <var>element</var>
-                    add the result of invoking this algorithm recursively
-                    passing <var>item</var> for <var>element</var>.
-                  </li>
-                </ol>
-              </li>
-              <li>Otherwise, if <var>element</var> is a <a>map</a>:
+              <li>Deserialize <var>key</var> into a new
+                <a>map</a> <var>embedded</var>.</li>
```
Comment (pchampin) [inline index.html:1424]: Why not simply getting the `@id` entry of <var>annotation</var>? Shouldn't this be equivalent?

File: index.html [hunk 2]
```diff
@@ -1341,43 +1363,89 @@ <h4>Rename Blank Nodes</h4>
...
+              <li>Initialize <var>id</var> with the value of the `@id`
+                entry from embedded, using the
+                <a data-cite="JSON-LD11-API#dfn-canonical-lexical-form">canonical lexical form</a>
+                if it is a <a>map</a>.</li>
+              <li>Initialize <var>node</var> to the entry for <var>id</var>
+                in <var>node map</var>. If there is no such <a>entry</a>,
+                continue to the next <a>entry</a> in <var>node map</var>.</li>
+              <li>Initialize <var>property</var> and <var>value</var>
+                to the first value
+                from the reamining <a>entry</a> from <var>embedded</var>.</li>
```
Comment (pchampin) [inline index.html:1434]:
```suggestion
              <li>Besides `@id`, <var>embedded</var> must have exactly one <a>entry</a>. Initialize <var>property</var> and <var>value</var>
                to the key and the first value
                from that entry <a>entry</a>.</li>
```

File: index.html [hunk 3]
```diff
@@ -1341,43 +1363,89 @@ <h4>Rename Blank Nodes</h4>
...
-                      <li>Otherwise, add an <a>entry</a> to <var>result</var> for `@id`
-                        with the value from the result of calling this algorithm recursively
-                        using <var>value</var> for <var>element</var>.</li>
-                    </ol>
-                  </li>
+                  <li>Remove the <var>key</var> <var>entry</var>
+                    from <var>node map</var>.</li>
+                  <li>Remove the `@id` <var>entry</var>
+                    from <var>annotation</var>.</li>
+                  <li>Use <var>add value</var> to the property <var>property</var> in <var>node</var>, using `true` for <em>as array</em>.</li>
```
Comment (pchampin) [inline index.html:1442]:
```suggestion
                  <li>Use <var>add value</var> to add <var>annotation</var> to the property <var>property</var> in <var>node</var>, using `true` for <em>as array</em>.</li>
```

**Review by pchampin — APPROVED** [review]

File: index.html (hunk 1 above)
Comment (gkellogg) [reply index.html:1424]: Yes, probably. I'll verify in my own code base.

File: index.html (hunk 2 above)
Comment (gkellogg) [reply index.html:1434]: This is good, but I'm going to tweak the language a little bit.

File: index.html (hunk 1 above)
Comment (gkellogg) [reply index.html]: Addressed in 311fce8e.

File: index.html (hunk 2 above)
Comment (gkellogg) [reply index.html]: Addressed in 311fce8e.

File: index.html [hunk 4]
```diff
@@ -1341,43 +1363,90 @@ <h4>Rename Blank Nodes</h4>
...
-                      <li>Otherwise, add an <a>entry</a> to <var>result</var> for `@id`
-                        with the value from the result of calling this algorithm recursively
-                        using <var>value</var> for <var>element</var>.</li>
-                    </ol>
-                  </li>
+                  <li>Remove the <var>key</var> <a>entry</a>
+                    from <var>node map</var>.</li>
+                  <li>Remove the `@id` <var>entry</var>
+                    from <var>annotation</var>.</li>
+                  <li>Use <var>add value</var> to add <var>annotation</var> to the property <var>property</var> in <var>node</var>, using `true` for <em>as array</em>.</li>
```
Comment (pchampin) [inline index.html:1443]:
```suggestion
                  <li>Use <a data-cite="JSON-LD11-API#dfn-add-value">add value</a> to add <var>annotation</var> to the property <var>property</var> in <var>node</var>, using `true` for <em>as array</em>.</li>
```

**Review by pchampin — APPROVED** [review]
Still a little fix required, but overall good to go :)

**Pull request merged**